File: easybluetoothble/src/main/java/com/srise/easybluetoothble/Interface/IBleGattService.java
package com.srise.easybluetoothble.Interface;

import android.bluetooth.BluetoothGattCharacteristic;
import android.bluetooth.BluetoothGattService;

import java.util.Map;


public interface IBleGattService {
    /**
     * 获取service
     *
     * @return
     */
    BluetoothGattService getGattService();

    /**
     * 获取service UUID
     *
     * @return
     */
    String getGattServiceUUID();


    /**
     * 获取对应service的dispatcher
     *
     * @return
     */
    Class<?> getDispatcher();

    /**
     * 获取对应service的BluetoothGattCharacteristic map
     *
     * @return
     */
    Map<String, BluetoothGattCharacteristic> getCharacteristicMap();
}

File: ktlionble/src/main/java/c/example/dell/ktlionble/inter/IBleGattService.kt
package c.example.dell.ktlionble.inter

import android.bluetooth.BluetoothGattCharacteristic
import android.bluetooth.BluetoothGattService

interface IBleGattService {
    fun getCharacteristicMap(): Map<out String, BluetoothGattCharacteristic>

    fun getDispatcher() : Class<*>

    fun getGattService(): BluetoothGattService?

    fun getGattServiceUUID(): String?
}